### ISSUE TITLE
Test: IE (9, 10, and 11) returns different cache hash fn than other browsers

### ISSUE THREAD

**Issue by rxaviers** [jquery/globalize#573]
Functional tests show that IE (9, 10, and 11) uses a different hash value than other browsers, which would break runtime compiled applications;  (Note Edge 12 works fine) 

Interestingly, I have tested the compiled app of example https://github.com/jquery/globalize/tree/master/examples/app-npm-webpack and it works just fine on all these browsers. So, is it a test issue?

**Comment by rxaviers** [comment]
Example: (the same happens for IE9, 10, and 11).
![bs_win7_ie_10 0](https://cloud.githubusercontent.com/assets/967155/12171708/2ff36ef8-b533-11e5-924c-8b47af2f60df.jpg)

**Comment by rxaviers** [comment]
Somewhat related #572

**Comment by alunny** [comment]
fwiw, I am seeing requirejs issues with running the functional tests in browserstack's release of IE.

![screen shot 2016-01-09 at 10 24 27 pm](https://cloud.githubusercontent.com/assets/48361/12220253/d1fca0fa-b71f-11e5-97b4-453482873eef.png)

Not sure if that is a known issue or not.

**Comment by alunny** [comment]
Okay, I believe I know the problem.

`src/common/runtime-bind.js` uses `fn.name` to generate the hash key: https://github.com/jquery/globalize/blob/master/src/common/runtime-bind.js#L9

Function.name is not supported in IE*, and so returns undefined.

A patch like the following in `src/number.js` (and equivalent in the other modules) seems to fix the tests in IE:

```diff
+ if ( !returnFn.name ) {
+    returnFn.name = 'numberParser';
+ }
```

That said, it might be wiser to write to a non-standard property (like `Function.displayName`) and always use that, to avoid any potential issues with writing to a non-writable property in environments that do support Function.name.

\* per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name not my own testing

**Comment by rxaviers** [comment]
@scottgonzalez @jzaefferer any suggestions please?

PS: When coding I had initially came up with https://github.com/jquery/globalize/commit/a03a92db431a0d2c9e42ed3b24c8654ce490e200, then changed it to `fn.name` after having forgot what the complicated code was for.

http://stackoverflow.com/a/6903860/798133

**Comment by rxaviers** [comment]
@alunny thanks for digging into it.

**Comment by scottgonzalez** [comment]
Seems like `var fnName = fn.name || /^function\s+([\w\$]+)\s*\(/.exec( fn.toString() )[ 1 ];` with a support comment would be fine.

**Comment by rxaviers** [comment]
@scottgonzalez thanks

**Comment by rxaviers** [comment]
Fixed by 831602977d12496f886f03fce63082669828ae0f
